File: android/app/src/main/java/sol/vortexx/android/crypto/api/Signer.kt
package sol.vortexx.android.crypto.api

/**
 * Digital signature over a detached message (Ed25519 by default).
 *
 * Two operations only — sign and verify. Key generation stays here too
 * because the curve + encoding are implementation-private (Ed25519 uses
 * raw 32/32, ECDSA would be DER, etc.).
 */
interface Signer {
    fun generateKeyPair(): KeyPair
    fun sign(privateKey: ByteArray, message: ByteArray): ByteArray
    fun verify(publicKey: ByteArray, message: ByteArray, signature: ByteArray): Boolean
}

File: vortex_waf/src/ports/signer.rs
//! Подпись и её проверка в постоянном времени.

pub trait Signer: Send + Sync {
    fn sign(&self, payload: &str) -> String;

    /// Сравнение обязано быть устойчивым к атакам по времени.
    fn verify(&self, payload: &str, signature: &str) -> bool;
}
